### PR TITLE
Added order statistics to the processor statistics plugin and updated the unit t…

### PR DESCRIPTION
**Pull request by rashmigottipati**
Summary of changes:

Added order statistics to the processor plugin. Updated to the new go library. 

How to verify it:
make test-small

Testing done:
make test-small

### CODE REVIEW

File: statistics/statistics.go [hunk 1]
```diff
@@ -19,28 +19,21 @@ limitations under the License.
...
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
 	"github.com/montanaflynn/stats"
 )
 
 const (
-	pluginName = "statistics"
-	version    = 1
-	pluginType = plugin.ProcessorPluginType
+	Name    = "statistics"
+	Version = 1
```
Comment (kindermoumoute) [inline statistics/statistics.go]: Update version to 2

File: statistics/statistics.go [hunk 2]
```diff
@@ -156,17 +147,24 @@ func (p *Plugin) calculateStats(buff interface{}, startTime time.Time, stopTime
...
-			Data_:               val,
-			Namespace_:          core.NewNamespace(namespace, stat),
-			Timestamp_:          time,
-			LastAdvertisedTime_: time,
-			Unit_:               unit,
-			Tags_:               tags,
+		metric := plugin.Metric{
+			Data:      val,
+			Namespace: plugin.NewNamespace(namespace, stat),
+			Timestamp: time,
```
Comment (candysmurf) [inline statistics/statistics.go]: why did you remove LastAdvertisedTime? would you please add it back in?

File: statistics/statistics.go [hunk 3]
```diff
@@ -310,79 +355,65 @@ func (p *Plugin) getTimes(times []time.Time) (time.Time, time.Time) {
...
+			p.insertInToBuffer(metric.Data, metric.Namespace.String())
 			p.updateCounters()
 
 			if p.bufferCurSize < p.bufferMaxSize {
-				log.Printf("Buffer: %v", p.buffer[k])
 				stats, err := p.calculateStats(p.buffer[k][0:p.bufferCurSize], startTime, stopTime, k, unit)
 				if err != nil {
-					log.Printf("Error occured in calculating Statistics: %s", err)
-					return "", nil, err
+					log.Warnf("Error occured in calculating Statistics: %s", err)
```
Comment (candysmurf) [inline statistics/statistics.go]: any place returns an error, let's log them as Errorf.
